homeassistant/components/enphase_envoy/config_flow.py

```diff
@@ -287,7 +287,7 @@ async def async_step_reconfigure(
             await self.async_set_unique_id(entry.unique_id)
 
         self.context["title_placeholders"] = {
-            CONF_SERIAL: self.unique_id,
+            CONF_SERIAL: self.unique_id or "-",
```

Comment:
We should already have a unique id at this point because of the lines above, right?

Reply:
No, there's no such guarantee. There's nothing preventing a user flow from adding a device which doesn't have a serial number.